runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuNativeWindow.java

```diff
@@ -0,0 +1,216 @@
+				{
+					listener.mouseExited(event);
+				}
+			}
+
+			@Override
+			public void mousePressed(MouseEvent e)
+			{
+				final long now = System.currentTimeMillis();
+				final java.awt.event.MouseEvent event = new java.awt.event.MouseEvent(
+					canvas, java.awt.event.MouseEvent.MOUSE_PRESSED, now, 0, e.getX(), e.getY(), 1, false,
+					e.getButton());
+
+				for (java.awt.event.MouseListener listener : canvas.getMouseListeners())
+				{
+					listener.mousePressed(event);
+				}
+			}
+
+			@Override
+			public void mouseReleased(MouseEvent e)
+			{
+				final long now = System.currentTimeMillis();
+				final java.awt.event.MouseEvent event = new java.awt.event.MouseEvent(
+					canvas, java.awt.event.MouseEvent.MOUSE_RELEASED, now, 0, e.getX(), e.getY(), 1, false,
+					e.getButton());
+
+				for (java.awt.event.MouseListener listener : canvas.getMouseListeners())
+				{
+					listener.mouseReleased(event);
+				}
+			}
+
+			@Override
+			public void mouseMoved(MouseEvent e)
+			{
+				final long now = System.currentTimeMillis();
+				final java.awt.event.MouseEvent event = new java.awt.event.MouseEvent(
+					canvas, java.awt.event.MouseEvent.MOUSE_MOVED, now, 0, e.getX(), e.getY(), 1, false,
+					e.getButton());
+
+				for (java.awt.event.MouseMotionListener listener : canvas.getMouseMotionListeners())
+				{
+					listener.mouseMoved(event);
+				}
+			}
+
+			@Override
+			public void mouseDragged(MouseEvent e)
+			{
+				final long now = System.currentTimeMillis();
+				final java.awt.event.MouseEvent event = new java.awt.event.MouseEvent(
+					canvas, java.awt.event.MouseEvent.MOUSE_DRAGGED, now, 0, e.getX(), e.getY(), 1, false,
+					e.getButton());
+
+				for (java.awt.event.MouseMotionListener listener : canvas.getMouseMotionListeners())
+				{
+					listener.mouseDragged(event);
+				}
+			}
+
+			@Override
+			public void mouseWheelMoved(MouseEvent e)
+			{
+
+			}
+		});
+
+		window.addKeyListener(new KeyListener()
+		{
+			@Override
+			public void keyPressed(KeyEvent e)
+			{
+				final long now = System.currentTimeMillis();
+				final java.awt.event.KeyEvent eventTyped = new java.awt.event.KeyEvent(
+					canvas, java.awt.event.KeyEvent.KEY_TYPED,
+					now, 0, KeyEvent.VK_UNDEFINED, e.getKeyChar());
+				final java.awt.event.KeyEvent event = new java.awt.event.KeyEvent(
+					canvas, java.awt.event.KeyEvent.KEY_PRESSED,
+					now, 0, AWTNewtEventFactory.newtKeyCode2AWTKeyCode(e.getKeyCode()), e.getKeyChar());
+
+				for (java.awt.event.KeyListener listener : canvas.getKeyListeners())
+				{
+					listener.keyTyped(eventTyped);
+					listener.keyPressed(event);
+				}
+			}
+
+			@Override
+			public void keyReleased(KeyEvent e)
+			{
+				final long now = System.currentTimeMillis();
+				final java.awt.event.KeyEvent event = new java.awt.event.KeyEvent(
+					canvas, java.awt.event.KeyEvent.KEY_RELEASED,
+					now, 0, e.getKeyCode(), e.getKeyChar());
```

Comment:
```suggestion
					now, 0, AWTNewtEventFactory.newtKeyCode2AWTKeyCode(e.getKeyCode()), e.getKeyChar());
```

This fixes an issue where keyRelease events never worked and testing this, I would get bad things happening like the camera forever rotating and not recognizing I only pressed the arrow key once and released already.